truffle/src/main/java/org/jruby/truffle/core/string/StringNodes.java

```diff
@@ -3103,6 +3123,19 @@ protected static boolean hasRawBytes(DynamicObject string) {
 
     }
 
+    @CoreMethod(names = "string_escape")
```

Comment:
Make this a primitive, like the node just below.
You can then invoke it with `Truffle.invoke_primitive :string_escape, str`.
This avoids polluting String's namespace.